apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.cs

```diff
@@ -1520,78 +1568,70 @@ void RetryAcksAndExtends(bool? ack, AckError ackError, IEnumerable<TimedId> exte
                     }
 
                     var idsToRetry = hasAcksOrNacks ? ackOrNackIds : extendIds.Select(j => j.Id);
-                    var retryableErrorIds = new ConcurrentBag<string>();
+                    var retryableErrorIds = new List<string>();
                     // Not all temporaryFailures may need to be retried.
                     // See if the id exists in the _retryableIds dictionary and only then retry.
-                    foreach (var id in idsToRetry.Where(j => _retryableIds.ContainsKey(j)))
-                    {
-                        retryableErrorIds.Add(id);
-                    }
+                    retryableErrorIds.AddRange(idsToRetry.Where(j => _retryableIds.ContainsKey(j)));
```

Comment:
Now this is a list, we can just have:

```csharp
var retryableErrorIds = idsToRetry.Where(j => _retryableIds.ContainsKey(j)).ToList();
```

Or potentially:

```csharp
var retryableErrorIds = idsToRetry.Intersect(_retryableIds.Keys).ToList();
```

I don't know whether the latter would create a snapshot though.

Reply:
Done. Used the version with Where.